fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -296,17 +296,7 @@ ThreadResult<RangeResult> DLTransaction::readBlobGranules(const KeyRangeRef& key
 	                                                         beginVersion,
 	                                                         rv,
 	                                                         context);
-	const FdbCApi::FDBKeyValue* kvs;
-	int count;
-	FdbCApi::fdb_bool_t more;
-	FdbCApi::fdb_error_t error = api->resultGetKeyValueArray(r, &kvs, &count, &more);
-	if (error) {
-		return ThreadResult<RangeResult>(Error(error));
-	}
-
-	// The memory for this is stored in the FDBResult and is released when the result gets destroyed
-	return ThreadResult<RangeResult>(
-	    RangeResult(RangeResultRef(VectorRef<KeyValueRef>((KeyValueRef*)kvs, count), more), Arena()));
+	return ThreadResult<RangeResult>((ThreadSingleAssignmentVar<RangeResult>*)(r));
```

Comment:
I'm pretty sure this is undefined behavior, since ThreadSingleAssignmentVar will technically be a different type (it's not exposed by the dlopen'd libfdb_c). This probably works ok in practice, but it's possible that we change the ABI of ThreadSingleAssignmentVar and then this stops working.

Reply:
Let's not let this block 7.1, since this won't manifest until we load this client as an external library from a future libfdb_c.so. As long as we fix it properly before the next release this shouldn't manifest.